akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/StashSpec.scala

```diff
@@ -562,26 +562,105 @@ class UnstashingSpec extends ScalaTestWithActorTestKit("""
         val stash = StashBuffer[String](10)
         stash.stash("one")
 
-        // FIXME #26148 using AbstractBehavior because unstashAll doesn't support Behavior.same
-
         // unstashing is inside setup
-        new AbstractBehavior[String] {
-          override def onMessage(msg: String): Behavior[String] = msg match {
-            case "unstash" =>
-              Behaviors.setup[String] { ctx =>
-                stash.unstashAll(ctx, this)
-              }
-            case _ =>
-              probe.ref ! msg
-              Behavior.same
-          }
+        Behaviors.receiveMessage {
+          case "unstash" =>
+            Behaviors.setup[String] { ctx =>
+              stash.unstashAll(ctx, Behaviors.same)
+            }
+          case msg =>
+            probe.ref ! msg
+            Behavior.same
         }
       })
 
       ref ! "unstash"
       probe.expectMessage("one")
     }
 
-  }
+    "deal with unhandled the same way as normal unhandled" in {
+      val ref = spawn(Behaviors.setup[String] { ctx =>
+        val stash = StashBuffer[String](10)
+        stash.stash("one")
+
+        Behaviors.receiveMessage {
+          case "unstash" =>
+            stash.unstashAll(ctx, Behaviors.receiveMessage {
```

Comment:
Just curious why we went with `Behaviors.receiveMessage` versus `Behaviors.receive` for `Actor.receive` and to not be explicit or conflict with message/event.

Reply:
I think `receive` just followed along from the untyped actor API. `receiveMessage` was added much later just to not have to deal with the context on every message unless you want to (not sure why it isn't just an overload though, it's `Function1` vs. `Function2` so erasure shouldn't be a problem, maybe something with type inference) 

In general I think the signals will be used much much less often so it's probably not _that_ bad that it is a bit inconsistent.